Polyfills/Canvas/Source/Shaders/fs_boxblur.sc

```diff
@@ -0,0 +1,48 @@
+$input v_position, v_texcoord0
+
+#include "./common.sh"
+
+uniform vec4 u_viewSize;	// vec4 (width, height, unused, unused)
+uniform vec4 u_direction;	// vec4 (x, y, unused, unused)
+uniform vec4 u_weights;		// vec4 (kernel, radius, offsetX, offsetY)
+
+SAMPLER2D(s_tex, 0);
+
+#if NEED_HALF_TEXEL
+uniform vec4 u_halfTexel;
+#endif // NEED_HALF_TEXEL
+
+#define kernel u_weights.x
+#define radius u_weights.y
+#define offsetX u_weights.z
+#define offsetY u_weights.w
+
+void main()
+{
+#if !NEED_HALF_TEXEL
+	vec4 u_halfTexel = vec4_splat(0.0);
+#endif // !NEED_HALF_TEXEL
+	vec2 texcoord0 = v_texcoord0 + u_halfTexel.xy;
+	vec4 color = vec4_splat(0.0);
+
+	// sample center if no offset (ie. odd kernel)
+	if (offsetX == 0.0f && offsetY == 0.0f)
```

Comment:
avoid comparing floats with literal values. check if value is absolute value is great than an epsilon. It should not be a problem if uniform set from C++ is a literal. but if it's the result of a computation, it might be something like 1.23e-6 which is not 0.